cmd/ctl/pkg/upgrade/migrate/command.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migrate
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+	apiextinstall "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/install"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/jetstack/cert-manager/cmd/ctl/pkg/build"
+	"github.com/jetstack/cert-manager/cmd/ctl/pkg/factory"
+	acmeinstall "github.com/jetstack/cert-manager/internal/apis/acme/install"
+	cminstall "github.com/jetstack/cert-manager/internal/apis/certmanager/install"
+)
+
+var (
+	long = templates.LongDesc(i18n.T(`
+Ensures resources in your Kubernetes cluster are persisted in the v1 API version.
+
+This must be run prior to upgrading to ensure your cluster is ready to upgrade to cert-manager v1.7 and beyond.
+
+This command must be run with a cluster running cert-manager v1.0 or greater.`))
+
+	example = templates.Examples(i18n.T(build.WithTemplate(`
+# Check the cert-manager installation is ready to be upgraded to v1.7
+{{.BuildName}} upgrade migrate
+`)))
+)
+
+// Options is a struct to support renew command
+type Options struct {
+	genericclioptions.IOStreams
+	*factory.Factory
+
+	client client.Client
+	force  bool
```

Comment:
I am still a little unconvinced about the flag name. Could we maybe make it so that it conveys what it does better? Perhaps just `--all`?

Reply:
Perhaps `--skip-stored-version-check`?

Reply:
> Perhaps --skip-stored-version-check?

I would prefer this to `--force` 👍🏼 I also like `--all` or something like that, that would convey the idea that 'all' will be updated and does not require users to think about what stored version check is .
However, it is always hard to agree on naming 😄

Reply:
My concern with `--all` is that it implies that if you *don't* use it, all things may *not* be migrated (when in fact they *may* still all be migrated by the end of running the command).

Given the target audience for this flag are those that have already manually edited the `status.storedVersions` field, they are likely to understand/recognise the language used in the flag name & description IMO

Reply:
`-all` doesn't immediately jump out as "something I shouldn't use" as a 99% user, whereas `--skip-stored-version-check` or `--force` sound like you are "messing with the process" and wouldn't be something I'd rush to use

Reply:
>  if you don't use it, all things may not be migrated 

But this is exactly why that flag is there though (although only in tiny number of edge cases)